applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+    QSVMS<TElementData>(NewId,pGeometry,pProperties)
+{}
+
+///////////Destructor
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::~QSVMSDEMCoupled()
+{}
+
+template< class TElementData >
+Element::Pointer QSVMSDEMCoupled<TElementData>::Create(IndexType NewId,NodesArrayType const& ThisNodes,Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+}
+
+template< class TElementData >
+Element::Pointer QSVMSDEMCoupled<TElementData>::Create(IndexType NewId,GeometryType::Pointer pGeom,Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, pGeom, pProperties);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template< class TElementData >
+int QSVMSDEMCoupled<TElementData>::Check(const ProcessInfo &rCurrentProcessInfo)
+{
+    int out = QSVMS<TElementData>::Check(rCurrentProcessInfo);
+    KRATOS_ERROR_IF_NOT(out == 0)
+        << "Error in base class Check for Element " << this->Info() << std::endl
+        << "Error code is " << out << std::endl;
+
+    // Extra variables
+    KRATOS_CHECK_VARIABLE_KEY(ACCELERATION);
+    KRATOS_CHECK_VARIABLE_KEY(NODAL_AREA);
+
+    // Output variables (for Calculate() functions)
+    KRATOS_CHECK_VARIABLE_KEY(SUBSCALE_VELOCITY);
+    KRATOS_CHECK_VARIABLE_KEY(SUBSCALE_PRESSURE);
+
+    for(unsigned int i=0; i<NumNodes; ++i)
+    {
+        Node<3>& rNode = this->GetGeometry()[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,rNode);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_AREA,rNode);
+    }
+
+    return out;
+}
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::Calculate(const Variable<double>& rVariable,
+                                             double& rOutput, 
+                                             const ProcessInfo& rCurrentProcessInfo) {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+
+        if (rVariable == ERROR_RATIO)
+        {
+            // Get the element's geometric parameters
+            double Weight = data.Weight;
+            // array_1d<double, TNumNodes> N;
+            // BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+            // GeometryUtils::CalculateGeometryData(this->GetGeometry(), data.DN_DX, data.N, data.Weight);
+            // Calculate this element's fluid properties
+            const double density = this->GetAtCoordinate(data.Density,data.N);
+
+            this->GetEffectiveViscosity(data, data.EffectiveViscosity);
+            const double viscosity = this->GetAtCoordinate(data.EffectiveViscosity,data.N);
+
+            // Get Advective velocity
+            array_1d<double, 3> convective_velocity= this->GetAtCoordinate(data.Velocity, data.N) -
+            this->GetAtCoordinate(data.MeshVelocity, data.N);
```

Comment:
Shouldn't you substract the `data.MeshVelocity` to the `convective_velocity` value?

Reply:
OK. Now I see. Forget about my coment.

Reply:
In any case, this can be a `const` array.